04-onion-routing.md

```diff
@@ -1007,6 +1027,22 @@ the decrypted byte stream.
 The complete amount of the multi-part payment was not received within a
 reasonable time.
 
+1. type: UPDATE|25 (`upfront_fee_insufficient`)
+2. data:
+   * [`u64`:`upfront_fee`]
+   * [`u16`:`len`]
+   * [`len*byte`:`channel_update`]
+
+The upfront fee amount was below that required by the channel from the
+processing node.
+
+1. type: 27 (`final_incorrect_upfront_fee`)
```

Comment:
Can't `upfront_fee_insufficient` be used here? The sender already knows which node the error originates from.

Reply:
Went with the same approach as we have for success-case fees here, we don't have a channel update so separated the error our.

Reply:
I thought a zero length chan update could already signal this. It seems two error types is redundant.